view/src/storage.rs

```diff
@@ -235,6 +235,37 @@ impl Storage {
         }
     }
 
+    /// Query for a swap by its swap commitment, optionally waiting until the note is detected.
+    pub fn swap_by_commitment(
+        &self,
+        swap_commitment: tct::Commitment,
+        await_detection: bool,
+    ) -> impl Future<Output = anyhow::Result<SwapRecord>> {
+        // Clone the pool handle so that the returned future is 'static
+        let pool = self.pool.clone();
+        async move {
+            // Check if we already have the note
+            if let Some(record) = sqlx::query_as::<_, SwapRecord>(
+                format!(
+                    "SELECT * FROM swaps WHERE swaps.swap_commitment = x'{}'",
+                    hex::encode(swap_commitment.0.to_bytes())
+                )
+                .as_str(),
+            )
+            .fetch_optional(&pool)
+            .await?
+            {
+                return Ok(record);
+            }
+
+            if !await_detection {
+                return Err(anyhow!("swap commitment {} not found", swap_commitment));
+            } else {
+                return Err(anyhow!("swap commitment await_detection not implemented"));
```

Comment:
Followup: implement awaiting detection of swap commitments in the view server.

This should work (and be implemented exactly like) detection for note commitments.  We don't currently need this in `pcli`, though, because we always await transaction confirmation anyways, and at that point the swap commitment we care about for claiming will already be there. But in the future this might be useful for someone else who's writing a more complex interaction (e.g., a trading bot).